Ginger/GingerCoreNET/GeneralLib/EmailReadMSGraphOperations.cs

```diff
@@ -67,36 +69,87 @@ public async Task ReadEmails(EmailReadFilters filters, EmailReadConfig config, A
             GraphServiceClient graphServiceClient, Action<ReadEmail> emailProcessor)
         {
             IEnumerable<string> expectedRecipients = null;
+            int  count = 0;
             if (!string.IsNullOrEmpty(filters.To))
             {
                 expectedRecipients = filters.To.Split(";", StringSplitOptions.RemoveEmptyEntries);
             }
-
-            foreach (ICollectionPage<Message> messageCollection in messageCollections)
+            try
             {
-                PageIterator<Message> messageIterator = PageIterator<Message>.CreatePageIterator(
-                    graphServiceClient,
-                    messageCollection,
-                    message =>
-                    {
-                        if (!DoesSatisfyToFilter(message, expectedRecipients))
-                        {
-                            return true;
-                        }
-                        if (!DoesSatisfyAttachmentFilter(graphServiceClient, message, filters).Result)
-                        {
-                            return true;
-                        }
-                        if (!DoesSatisfyBodyFilter(message, filters.Body))
+
+                foreach (ICollectionPage<Message> messageCollection in messageCollections)
+                {
+
+                    PageIterator<Message> messageIterator = PageIterator<Message>.CreatePageIterator(
+                        graphServiceClient,
+                        messageCollection,
+                        message =>
                         {
+                            if (!DoesSatisfyToFilter(message, expectedRecipients))
+                            {
+                                return true;
+                            }
+                            if (!DoesSatisfyAttachmentFilter(graphServiceClient, message, filters).Result)
+                            {
+                                return true;
+                            }
+                            if (!DoesSatisfyBodyFilter(message, filters.Body))
+                            {
+                                return true;
+                            }
+                            count++;
+                            if (count > filters.ReadCount)
```

Comment:
Please move this condition either after reading complete mail or at the start of this function to avoid executing other if conditions, also check if it actually not processing other emails once count is reached